metadb/store_test.go

```diff
@@ -0,0 +1,93 @@
+package metadb
+
+import (
+	"context"
+	"io/ioutil"
+	"os"
+	"sync"
+	"testing"
+
+	"github.com/pastelnetwork/gonode/common/log"
+	"github.com/stretchr/testify/suite"
+)
+
+func TestSuite(t *testing.T) {
+	suite.Run(t, new(testSuite))
+}
+
+type testSuite struct {
+	suite.Suite
+
+	config  *Config
+	service *Service
+	ctx     context.Context
+	cancel  context.CancelFunc
+	wg      sync.WaitGroup
+}
+
+func (s *testSuite) SetupSuite() {
+	// new the rqlite config
+	s.config = NewConfig()
+	// new the rqlite service
+	s.service = NewService(s.config)
+
+	// disable the logger
+	log.DefaultLogger.SetOutput(ioutil.Discard)
+	s.ctx, s.cancel = context.WithCancel(context.Background())
+
+	s.wg.Add(1)
+	go func() {
+		defer s.wg.Done()
+
+		// start the rqlite server
+		s.Nil(s.service.Run(s.ctx), "run service")
+	}()
+	// wait until the rqlite node is ready
+	<-s.service.ready
+}
+
+func (s *testSuite) TearDownSuite() {
+	if s.cancel != nil {
+		s.cancel()
+	}
+	s.wg.Wait()
+
+	// remove the data directy
+	os.RemoveAll(s.config.DataDir)
+}
+
+func (s *testSuite) TestWriteAndQuery() {
```

Comment:
We use [table driven tests](https://dave.cheney.net/2019/05/07/prefer-table-driven-tests) in all our unit tests 
please also use this package https://github.com/stretchr/testify
Examine other our unit tests to understand better.

Reply:
done